### PR TITLE
check for len(workers) in Scheduler.retire_workers

### PR DESCRIPTION
**Pull request by slavoutich**
In some scenarios workers set in retire_workers function gets empty,
    which then causes an exception in set.union, which gets no
    arguments. This commit adds check for number of workers in that
    place.

### CODE REVIEW

File: distributed/scheduler.py [hunk 1]
```diff
@@ -1791,8 +1791,11 @@ def retire_workers(self, comm=None, workers=None, remove=True, close=False,
                         pass
 
             workers = set(workers)
-            keys = set.union(*[self.has_what[w] for w in workers])
-            keys = {k for k in keys if self.who_has[k].issubset(workers)}
+            if len(workers) > 0:
+                keys = set.union(*[self.has_what[w] for w in workers])
+                keys = {k for k in keys if self.who_has[k].issubset(workers)}
+            else:
+                keys = {}
```
Comment (mrocklin) [inline distributed/scheduler.py]: Note that the literal `{}` is a dict.  You may want `set()` instead.

Comment (slavoutich) [inline distributed/scheduler.py]: Ah yeah, sorry, will fix in a minute.

Comment (slavoutich) [inline distributed/scheduler.py]: Running tests now, if Ok -- will push with test also.

**Comment by mrocklin** [comment]
I think that all this is missing is a small test.  @slavoutich was the comment above enough information to help you with this?

**Comment by mrocklin** [comment]
> Sorry for delay, at the conference now. Should be fine now.

Thank you for following through.  I've modified the test slightly and pushed.  Will merge when tests pass.

**Comment by mrocklin** [comment]
This is in.  Thanks @slavoutich !

**Pull request merged**